### PR TITLE
Profile icon/cdw24

### PR DESCRIPTION
**Pull request by mrgiles1**

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import { useEffect, useState } from "react";
 import { getAllUsers } from "./requests/getAllUsers";
-import { userData } from "./requests/types";
+import { userData } from "./theme/types";
```
Comment (jtabba) [inline frontend/src/App.tsx]: Move back. Types for requests are not relevant to themes.

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -26,19 +27,21 @@ const App = () => {
...
+
+	return (
+		<Router>
+			<Routes>
+				<Route path="/" element={<UserInterface />}>
+					<Route path="home" element={<Home />} />
+					<Route path="explore" element={<h1>Explore</h1>} />
+					<Route path="products" element={<h1>Products</h1>} />
+					<Route path="mentors" element={<h1>Mentors</h1>} />
+					<Route path="profile" element={<Profile />} />
```
Comment (jtabba) [inline frontend/src/App.tsx:39]: Change your formatter to use smaller tab width

File: frontend/src/components/ProfileDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  MenuList,
+  MenuItem,
+  Button,
+} from '@chakra-ui/react';
+
+interface ProfileDropdownProps {
+  onSignOut: () => void;
+}
+
+const ProfileDropdown: React.FC<ProfileDropdownProps> = ({onSignOut}) => {
```
Comment (jtabba) [inline frontend/src/components/ProfileDropdown.tsx]: Put a space between the ({ })

File: frontend/src/components/ProfileDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  const { user, isAuthenticated } = useAuth0();
+
+  const navigate = useNavigate();
+
+  if (!isAuthenticated) {
+    return null;
+  }
+
+  const goToProfile = () => {
```
Comment (jtabba) [inline frontend/src/components/ProfileDropdown.tsx]: Convention is to prefix function names with `handle....`

File: frontend/src/components/authentication/SignInOrOut.tsx [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
+import ProfileDropdown from "../ProfileDropdown";
+import React, { useEffect } from 'react';
 
 export const AuthButtons = () => {
 	const { loginWithRedirect, isAuthenticated, logout } = useAuth0();
 
+	useEffect(() => {
+		console.log("Authentication state changed: ", isAuthenticated);
+	  }, [isAuthenticated]);
+
```
Comment (jtabba) [inline frontend/src/components/authentication/SignInOrOut.tsx]: Remove

File: frontend/src/components/header/NavBar.tsx [hunk 6]
```diff
@@ -1,8 +1,7 @@
 import { Link } from "react-router-dom";
 
 export const NavBar = () => (
-	<>
-		<Link to={"/explore"}>Explore</Link>
+	<>	<Link to={"/explore"}>Explore</Link>
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: These should be on separate lines

File: frontend/src/requests/getAllUsers.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { isAxiosError } from "axios";
 import { AxiosClient } from "./clients/axiosClient";
-import { IErrorResponse, IResponse } from "./types";
+import { IErrorResponse, IResponse } from "../theme/types";
```
Comment (jtabba) [inline frontend/src/requests/getAllUsers.ts]: Move back

**Review by jtabba — CHANGES_REQUESTED** [review]
Great job overall. Left some comments

File: frontend/src/App.tsx (hunk 1 above)
Comment (mrgiles1) [reply frontend/src/App.tsx]: done. not sure how that happened

File: frontend/src/App.tsx (hunk 2 above)
Comment (mrgiles1) [reply frontend/src/App.tsx:39]: got it. 2 space for JS

File: frontend/src/components/ProfileDropdown.tsx (hunk 3 above)
Comment (mrgiles1) [reply frontend/src/components/ProfileDropdown.tsx]: done

File: frontend/src/components/ProfileDropdown.tsx (hunk 4 above)
Comment (mrgiles1) [reply frontend/src/components/ProfileDropdown.tsx]: good to know. Implemented

File: frontend/src/components/authentication/SignInOrOut.tsx (hunk 5 above)
Comment (mrgiles1) [reply frontend/src/components/authentication/SignInOrOut.tsx]: done

File: frontend/src/components/header/NavBar.tsx (hunk 6 above)
Comment (mrgiles1) [reply frontend/src/components/header/NavBar.tsx]: done

File: frontend/src/requests/getAllUsers.ts (hunk 7 above)
Comment (mrgiles1) [reply frontend/src/requests/getAllUsers.ts]: done

File: frontend/src/components/ProfileDropdown.tsx (hunk 3 above)
Comment (jtabba) [reply frontend/src/components/ProfileDropdown.tsx]: Outstanding task

**Review by jtabba — APPROVED** [review]

**Pull request merged**